src/components/Table/utils/TableState.ts

```diff
@@ -212,12 +212,13 @@ export class TableState {
   }
 
   setColumns(columns: GridColumnWithId<any>[], visibleColumnsStorageKey: string | undefined): void {
+    const isInitial = !this.columns || this.columns.length === 0;
     if (columns !== this.columns) {
       this.columns = columns;
       this.visibleColumnsStorageKey = visibleColumnsStorageKey ?? camelCase(columns.map((c) => c.id).join());
       this.visibleColumns.replace(readOrSetLocalVisibleColumnState(columns, this.visibleColumnsStorageKey));
       const expandedColumnIds = columns.filter((c) => c.initExpanded).map((c) => c.id);
-      this.expandedColumns.replace(expandedColumnIds);
+      if (isInitial) this.expandedColumns.replace(expandedColumnIds);
```

Comment:
Makes sense to me... don't blow away the existing `expandedColumns` set when updating, and instead keep the expanded states per what the table already has defined... I can't think of a reason why this would break anything 🤔

Reply:
I agree, great find @TylerR909 for such a targeted/small slice point to handle this, I would have thought it'd be more work.

Fwiw wouldn't mind a test just b/c I could see this pretty easily regressing (aka me breaking it :-P :-)) if/when we refactor TableState for whatever reasons in the future.